controllers/propagator/propagation.go

```diff
@@ -901,6 +906,74 @@ func (r *PolicyReconciler) processTemplates(
 			ManagedClusterName: decision.ClusterName,
 		}
 
+		// Handle value encryption initialization
+		usesEncryption := templates.UsesEncryption(
+			policyT.ObjectDefinition.Raw, templateCfg.StartDelim, templateCfg.StopDelim,
+		)
+		// Initialize AES Key and initialization vector
+		if usesEncryption && !templateCfg.EncryptionEnabled {
+			log.V(1).Info("Found an object definition requiring encryption. Handling encryption keys.")
+			// Get/generate the encryption key
+			encryptionKey, err := r.getEncryptionKey(decision.ClusterName)
+			if err != nil {
+				log.Error(err, "Failed to get/generate the policy encryption key")
+
+				return err
+			}
+
+			// Get/generate the initialization vector
+			initializationVector, err := r.getInitializationVector(
+				rootPlc.GetName(), decision.ClusterName, annotations,
+			)
+			if err != nil {
+				log.Error(err, "Failed to get initialization vector")
+
+				return err
+			}
+
+			// Set the initialization vector in the annotations
+			replicatedPlc.SetAnnotations(annotations)
+
+			// Set the EncryptionConfig with the retrieved key
+			templateCfg.EncryptionConfig = templates.EncryptionConfig{
+				EncryptionEnabled:    true,
+				AESKey:               encryptionKey,
+				InitializationVector: initializationVector,
+			}
+
+			err = tmplResolver.SetEncryptionConfig(templateCfg.EncryptionConfig)
+			if err != nil {
+				log.Error(err, "Error setting encryption configuration")
+
+				return err
+			}
+		}
+
+		// Set initialization vector annotation on the ObjectDefinition for the controller's use
+		policyTObjectUnstructured := &unstructured.Unstructured{}
+
+		jsonErr := json.Unmarshal(policyT.ObjectDefinition.Raw, policyTObjectUnstructured)
+		if jsonErr != nil {
+			// it shouldn't get here but if it did just log a msg
+			// its all right, a generic msg will be used on the managedcluster
+			log.Error(jsonErr, "Error unmarshalling the object definition to JSON")
+		} else {
+			policyTAnnotations := policyTObjectUnstructured.GetAnnotations()
+			if policyTAnnotations == nil {
+				policyTAnnotations = make(map[string]string)
+			}
+			if policyIV, ok := annotations[ivAnnotation]; ok {
```

Comment:
This if statement can be skipped once my previous comment about skipping this code if `usesEncryption` is false and `annotations[ivAnnotation]` can be assigned directly.